modules/xmpp/SDPUtil.js

```diff
@@ -443,6 +443,30 @@ var SDPUtil = {
             .split(" ")
             .map(ssrcStr => parseInt(ssrcStr));
     },
+
+    /**
+     * Sets the given codecName as the preferred codec by
+     *  moving it to the beginning of the payload types
+     *  list (modifies the given mline in place)
+     * @param {object} videoMLine the video mline object from
+     *  an sdp as parsed by transform.parse
+     * @param {string} the name of the preferred codec
+     */
+    preferVideoCodec: function(videoMLine, codecName) {
+        let payloadType = null;
```

Comment:
ok, i documented the behavior to prefer the first seen and added a test case to verify it.  being able to prefer something based on sub-parameters of the codec is actually quite a bit harder, as the parameters are in a single line and vary from one codec to another, so i think it'd be best to wait and see how we want to best differentiate between different sub-types of a single codec.